### PR TITLE
Add model training & prediction endpoints to the library

### PR DESCRIPTION
**Pull request by yonromai**
TL;DR:

```python
from nxontology_ml.data import get_efo_otar_slim
from nxontology_ml.model.predict import train_predict

nxo = get_efo_otar_slim()
df = train_predict(nxo=nxo)
```

(From readme)

Some notes:
* More context can be found in [this comment](https://github.com/related-sciences/nxontology-ml/issues/35#issuecomment-1740984286)
* The output of the prediction **should** be consistent when invoked from the same system (in new python session). From experience, it's likely to differ slightly across platform
* No file is written (responsibility of the caller), but the content of the data is identical to `data/efo_otar_slim_v3.57.0_precisions.tsv`
* FWIW, schemaless exports make me a little nervous - but I guess they seem to be the norm in the data-science world, so it's on me

### CODE REVIEW

File: nxontology_ml/model/predict.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+
+def train_predict(
+    conf: ModelConfig = DEFAULT_MODEL_CONFIG,
+    nxo: NXOntology[str] | None = None,
+    training_set: tuple[np.ndarray, np.ndarray] | None = None,
+    include_feature_values: bool = True,
+    train_take: int | None = 0,
+    predict_take: int | None = 0,
+) -> pd.DataFrame:
```
Comment (dhimmel) [inline nxontology_ml/model/predict.py:100]: Nice! This looks like the right arguments to expose.

Perhaps we add a docstring here with parameter descriptions. Specifically, not immediately sure what `train_take` and `predict_take` are doing.

**Review by dhimmel — APPROVED** [review]
No need for my re-review prior to merging. I'm looking forward to trying this out!

File: nxontology_ml/model/predict.py (hunk 1 above)
Comment (yonromai) [reply nxontology_ml/model/predict.py:100]: > Perhaps we add a docstring here with parameter descriptions. Specifically, not immediately sure what train_take and predict_take are doing.

That's a good point: `train_take` and `predict_take` allow to take a subset of the points for training and/or prediction to speed up the whole process (mostly useful when manually or automatically testing).

Will add the doc.

**Comment by dhimmel** [comment]
> FWIW, schemaless exports make me a little nervous - but I guess they seem to be the norm in the data-science world

I do not disagree. We often use pydantic internally, although the ability to apply it recordwise to a pandas dataframe is somewhat lacking. On the other hand, if this pipeline breaks, it's not a big deal as long as that results in failure and not bad output in a silent manner.
